File: setup_dto_files.py
import os
import pathlib

# Create the dto directory
dto_dir = pathlib.Path(r"C:\Users\jihen\Downloads\portfolio\src\main\java\jihen\portfolio\dto")
dto_dir.mkdir(parents=True, exist_ok=True)

# Create SkillNodeDto.java
skill_node_content = '''package jihen.portfolio.dto;

import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class SkillNodeDto {
    private Long id;
    private String name;
    private String category;
    private Boolean isTrendy;
    private String description;
    private Integer searchCount;
}
'''

skill_file = dto_dir / "SkillNodeDto.java"
skill_file.write_text(skill_node_content)
print(f"Created: {skill_file}")

# Create PortfolioDNADto.java
portfolio_dna_content = '''package jihen.portfolio.dto;

import lombok.*;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PortfolioDNADto {
    private Long portfolioId;
    private String title;
    private String bio;
    private Integer totalViews;
    private Integer followerCount;
    private Boolean isVerified;
    private Set<SkillNodeDto> skills;
    private Set<String> projectCategories;
    private String primaryFocus;
    private String expertise;
}
'''

portfolio_file = dto_dir / "PortfolioDNADto.java"
portfolio_file.write_text(portfolio_dna_content)
print(f"Created: {portfolio_file}")

# Verify
print(f"\nDirectory exists: {dto_dir.exists()}")
print(f"Files in directory:")
for f in dto_dir.iterdir():
    print(f"  - {f.name}")

File: SetupDTOFiles.java
import java.io.File;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;

public class SetupDTOFiles {
    public static void main(String[] args) throws Exception {
        String basePath = "C:\\Users\\jihen\\Downloads\\portfolio\\src\\main\\java\\jihen\\portfolio\\dto";
        
        // Create directory
        Files.createDirectories(Paths.get(basePath));
        System.out.println("✓ Created directory: " + basePath);
        
        // Create SkillNodeDto.java
        String skillNodeDto = "package jihen.portfolio.dto;\n\n" +
            "import lombok.*;\n\n" +
            "@Getter\n" +
            "@Setter\n" +
            "@AllArgsConstructor\n" +
            "@NoArgsConstructor\n" +
            "@Builder\n" +
            "public class SkillNodeDto {\n" +
            "    private Long id;\n" +
            "    private String name;\n" +
            "    private String category;\n" +
            "    private Boolean isTrendy;\n" +
            "    private String description;\n" +
            "    private Integer searchCount;\n" +
            "}\n";
        
        Files.write(Paths.get(basePath, "SkillNodeDto.java"), skillNodeDto.getBytes(StandardCharsets.UTF_8));
        System.out.println("✓ Created SkillNodeDto.java");
        
        // Create PortfolioDNADto.java
        String portfolioDNADto = "package jihen.portfolio.dto;\n\n" +
            "import lombok.*;\n" +
            "import java.util.Set;\n\n" +
            "@Getter\n" +
            "@Setter\n" +
            "@AllArgsConstructor\n" +
            "@NoArgsConstructor\n" +
            "@Builder\n" +
            "public class PortfolioDNADto {\n" +
            "    private Long portfolioId;\n" +
            "    private String title;\n" +
            "    private String bio;\n" +
            "    private Integer totalViews;\n" +
            "    private Integer followerCount;\n" +
            "    private Boolean isVerified;\n" +
            "    private Set<SkillNodeDto> skills;\n" +
            "    private Set<String> projectCategories;\n" +
            "    private String primaryFocus;\n" +
            "    private String expertise;\n" +
            "}\n";
        
        Files.write(Paths.get(basePath, "PortfolioDNADto.java"), portfolioDNADto.getBytes(StandardCharsets.UTF_8));
        System.out.println("✓ Created PortfolioDNADto.java");
        
        // Verify
        System.out.println("\n✓ Files created successfully!");
        System.out.println("Directory: " + basePath);
        System.out.println("Files:");
        new File(basePath).listFiles((d, n) -> {
            System.out.println("  - " + n);
            return true;
        });
    }
}
